<a href="https://colab.research.google.com/github/valeriafigueredo-data/modelos_preditivos_classicos/blob/main/1_exercicioEDA_Classifica%C3%A7%C3%A3o_treino_teste_titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Grupo Sala 01**
* Arthur Wanderley Ferreira dos Santos
* Thiago Wanderley Amorim
* Manuela de Lacerda Bezerra Carvalho
* Valéria Cristina Andrade Rodrigues de Figueredo
* André Felipe Bezerra de Souza Leão


# Análise Exploratória de Dados (EDA)

O objetivo deste estudo é aplicar transformações, normalizações e análises descritivas e estatísticas dos dados. Por fim, é preciso que seja feita uma conclusão do estudo realizado.

Base de dados utilizada: Titanic
1. Carregamento de dados
2. Análise dos dados
3. Transformação dos dados (categoricos/binarização)
2. Remoção de campos desnecessários (nome, id, ticket, cabine)
4. Tratamento de dados nulos e/ou faltantes
5. Separação dos dados, holdout
6. Normalização dos dados

## Importar bibliotecas
** Importe pandas, seaborn e as bibliotecas usuais. **

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

## Obtenha os dados
** Leia o arquivo csv 'titanic.csv' em um DataFrame **

In [ ]:
df = pd.read_csv('https://raw.githubusercontent.com/agconti/kaggle-titanic/master/data/train.csv')

** Verifique o cabeçalho do DataFrame.**

In [ ]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

# Transformação dos dados (categoricos/binarização)

## Categorização

In [ ]:
from sklearn import preprocessing

In [ ]:
le = preprocessing.LabelEncoder()

In [ ]:
le.fit(df['Sex'])

LabelEncoder()

In [ ]:
df['Sex_le'] = le.transform(df['Sex'])

In [ ]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  Sex_le  
0        0         A/5 21171   7.2500   NaN        S       1  
1        0          PC 17599  71.2833   C85        C       0  
2        0  STON/O2. 3101282   7.9250   NaN        S       0  
3        0            113803  53.1000  C123        S       0  
4        0            373450   8.0500   NaN        S       1  
..     ...               ...      ...   ...      ...     ...  
886      0            211536  13.0000   NaN        S       1  
887      0            112053  30.0000   B42        S       0  
888      2        W./C. 6607  23.4500   NaN        S       0  
889      0            111369  30.0000  C148        C       1  
890      0            370376   7.7500   NaN        Q       1  

[891 rows x 13 columns]

Na categoria Sex_le, male é 1 e female é 0.

In [ ]:
df['Embarked'].unique()

array(['S', 'C', 'Q', nan], dtype=object)

## Binarização

In [ ]:
df_dummies = pd.get_dummies(df['Embarked'], drop_first=False, prefix='Embarked').astype(int)

In [ ]:
df_embarked = pd.concat([df, df_dummies], axis=1)

In [ ]:
df_embarked.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 16 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
 12  Sex_le       891 non-null    int64  
 13  Embarked_C   891 non-null    int64  
 14  Embarked_Q   891 non-null    int64  
 15  Embarked_S   891 non-null    int64  
dtypes: float64(2), int64(9), object(5)
memory usage: 111.5+ KB


In [ ]:
df_embarked

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  Sex_le  Embarked_C  \
0        0         A/5 21171   7.2500   NaN        S       1           0   
1        0          PC 17599  71.2833   C85        C       0           1   
2        0  STON/O2. 3101282   7.9250   NaN        S       0           0   
3        0            113803  53.1000  C123        S       0           0   
4        0            373450   8.0500   NaN        S       1           0   
..     ...               ...      ...   ...      ...     ...         ...   
886      0            211536  13.0000   NaN        S       1           0   
887      0            112053  30.0000   B42        S       0           0   
888      2        W./C. 6607  23.4500   NaN        S       0           0   
889      0            111369  30.0000  C148        C       1           1   
890      0            370376   7.7500   NaN        Q       1           0   

     Embarked_Q  Embarked_S  
0             0           1  
1             0           0  
2             0           1  
3             0           1  
4             0           1  
..          ...         ...  
886           0           1  
887           0           1  
888           0           1  
889           0           0  
890           1           0  

[891 rows x 16 columns]

# Remoção de campos desnecessários (PassengerId, Name, Ticket, Cabin, Embarked)

In [ ]:
df_embarked.drop(columns=['PassengerId', 'Name','Ticket', 'Cabin', 'Embarked'], inplace=True)

In [ ]:
df_embarked

Survived  Pclass     Sex   Age  SibSp  Parch     Fare  Sex_le  \
0           0       3    male  22.0      1      0   7.2500       1   
1           1       1  female  38.0      1      0  71.2833       0   
2           1       3  female  26.0      0      0   7.9250       0   
3           1       1  female  35.0      1      0  53.1000       0   
4           0       3    male  35.0      0      0   8.0500       1   
..        ...     ...     ...   ...    ...    ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000       1   
887         1       1  female  19.0      0      0  30.0000       0   
888         0       3  female   NaN      1      2  23.4500       0   
889         1       1    male  26.0      0      0  30.0000       1   
890         0       3    male  32.0      0      0   7.7500       1   

     Embarked_C  Embarked_Q  Embarked_S  
0             0           0           1  
1             1           0           0  
2             0           0           1  
3             0           0           1  
4             0           0           1  
..          ...         ...         ...  
886           0           0           1  
887           0           0           1  
888           0           0           1  
889           1           0           0  
890           0           1           0  

[891 rows x 11 columns]

# Tratamento de dados nulos e/ou faltantes

Describe

In [ ]:
df_embarked.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Survived    891 non-null    int64  
 1   Pclass      891 non-null    int64  
 2   Sex         891 non-null    object 
 3   Age         714 non-null    float64
 4   SibSp       891 non-null    int64  
 5   Parch       891 non-null    int64  
 6   Fare        891 non-null    float64
 7   Sex_le      891 non-null    int64  
 8   Embarked_C  891 non-null    int64  
 9   Embarked_Q  891 non-null    int64  
 10  Embarked_S  891 non-null    int64  
dtypes: float64(2), int64(8), object(1)
memory usage: 76.7+ KB


Male

In [ ]:
df_embarked.loc[df_embarked.Age.isna() & (df_embarked.Pclass == 1) & (df_embarked.Sex == 'male'), 'Age'] = df_embarked[df_embarked.Pclass == 1 & (df_embarked.Sex == 'male')].Age.median()
df_embarked.loc[df_embarked.Age.isna() & (df_embarked.Pclass == 2) & (df_embarked.Sex == 'male'), 'Age'] = df_embarked[df_embarked.Pclass == 2 & (df_embarked.Sex == 'male')].Age.median()
df_embarked.loc[df_embarked.Age.isna() & (df_embarked.Pclass == 3) & (df_embarked.Sex == 'male'), 'Age'] = df_embarked[df_embarked.Pclass == 3 & (df_embarked.Sex == 'male')].Age.median()

Female

In [ ]:
df_embarked.loc[df_embarked.Age.isna() & (df_embarked.Pclass == 1) & (df_embarked.Sex == 'female'), 'Age'] = df_embarked[df_embarked.Pclass == 1 & (df_embarked.Sex == 'female')].Age.median()
df_embarked.loc[df_embarked.Age.isna() & (df_embarked.Pclass == 2) & (df_embarked.Sex == 'female'), 'Age'] = df_embarked[df_embarked.Pclass == 2 & (df_embarked.Sex == 'female')].Age.median()
df_embarked.loc[df_embarked.Age.isna() & (df_embarked.Pclass == 3) & (df_embarked.Sex == 'female'), 'Age'] = df_embarked[df_embarked.Pclass == 3 & (df_embarked.Sex == 'female')].Age.median()

Removendo os dados nulos remanencentes

In [ ]:
df_embarked.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Survived    891 non-null    int64  
 1   Pclass      891 non-null    int64  
 2   Sex         891 non-null    object 
 3   Age         880 non-null    float64
 4   SibSp       891 non-null    int64  
 5   Parch       891 non-null    int64  
 6   Fare        891 non-null    float64
 7   Sex_le      891 non-null    int64  
 8   Embarked_C  891 non-null    int64  
 9   Embarked_Q  891 non-null    int64  
 10  Embarked_S  891 non-null    int64  
dtypes: float64(2), int64(8), object(1)
memory usage: 76.7+ KB


In [ ]:
df_embarked = df_embarked.dropna(subset='Age')

In [ ]:
df_embarked.info()

<class 'pandas.core.frame.DataFrame'>
Index: 880 entries, 0 to 890
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Survived    880 non-null    int64  
 1   Pclass      880 non-null    int64  
 2   Sex         880 non-null    object 
 3   Age         880 non-null    float64
 4   SibSp       880 non-null    int64  
 5   Parch       880 non-null    int64  
 6   Fare        880 non-null    float64
 7   Sex_le      880 non-null    int64  
 8   Embarked_C  880 non-null    int64  
 9   Embarked_Q  880 non-null    int64  
 10  Embarked_S  880 non-null    int64  
dtypes: float64(2), int64(8), object(1)
memory usage: 82.5+ KB


In [ ]:
df_embarked.head()

Survived  Pclass     Sex   Age  SibSp  Parch     Fare  Sex_le  Embarked_C  \
0         0       3    male  22.0      1      0   7.2500       1           0   
1         1       1  female  38.0      1      0  71.2833       0           1   
2         1       3  female  26.0      0      0   7.9250       0           0   
3         1       1  female  35.0      1      0  53.1000       0           0   
4         0       3    male  35.0      0      0   8.0500       1           0   

   Embarked_Q  Embarked_S  
0           0           1  
1           0           0  
2           0           1  
3           0           1  
4           0           1

# Separação dos dados, holdout

Use o método train_test_split para dividir seus dados em um conjunto treino e teste.

Removendo a coluna Sex pois não será utilizada para treinar o modelo

In [ ]:
df_embarked.drop(columns=['Sex'], inplace=True)

<ipython-input-72-a07d2a175b96>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_embarked.drop(columns=['Sex'], inplace=True)


In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(df_embarked.drop(columns=['Survived']),
                                                    df_embarked['Survived'],
                                                    test_size=0.3,
                                                    random_state=111)

In [ ]:
pd.DataFrame(X_train)

Pclass   Age  SibSp  Parch      Fare  Sex_le  Embarked_C  Embarked_Q  \
689       1  15.0      0      1  211.3375       0           0           0   
135       2  23.0      0      0   15.0458       1           1           0   
97        1  23.0      0      1   63.3583       1           1           0   
25        3  38.0      1      5   31.3875       0           0           0   
424       3  18.0      1      1   20.2125       1           0           0   
..      ...   ...    ...    ...       ...     ...         ...         ...   
278       3   7.0      4      1   29.1250       1           0           1   
87        3  40.0      0      0    8.0500       1           0           0   
735       3  28.5      0      0   16.1000       1           0           0   
887       1  19.0      0      0   30.0000       0           0           0   
863       3  35.0      8      2   69.5500       0           0           0   

     Embarked_S  
689           1  
135           0  
97            0  
25            1  
424           1  
..          ...  
278           0  
87            1  
735           1  
887           1  
863           1  

[616 rows x 9 columns]

# Normalização dos dados

Hora de para padronizar as variáveis.

Import StandardScaler do Scikit-learn.

In [ ]:
from sklearn.preprocessing import StandardScaler

In [ ]:
scaler = StandardScaler()

Converta as variáveis independentes com o StandardScaler

In [ ]:
columns_to_scale = ['Age', 'Fare']
scaler.fit(X_train[columns_to_scale])

StandardScaler()

Use o método .transform () para transformar os parâmetros em uma versão padronizada.

In [ ]:
X_train[['Age_scaled', 'Fare_scaled']] = scaler.transform(X_train[columns_to_scale])

In [ ]:
X_train_fare_scaled = np.array(X_train['Fare_scaled'])

In [ ]:
X_train_fare_scaled[:10]

array([ 3.58581199, -0.34946131,  0.61911195, -0.02184146, -0.24587885,
       -0.50616919, -0.50575219, -0.49313793, -0.50984601, -0.49280514])

In [ ]:
scaler.fit(X_test[['Age', 'Fare']])

StandardScaler()

In [ ]:
X_test[['Age_scaled', 'Fare_scaled']] = scaler.transform(X_test[['Age', 'Fare']])

In [ ]:
X_train

Pclass   Age  SibSp  Parch      Fare  Sex_le  Embarked_C  Embarked_Q  \
689       1  15.0      0      1  211.3375       0           0           0   
135       2  23.0      0      0   15.0458       1           1           0   
97        1  23.0      0      1   63.3583       1           1           0   
25        3  38.0      1      5   31.3875       0           0           0   
424       3  18.0      1      1   20.2125       1           0           0   
..      ...   ...    ...    ...       ...     ...         ...         ...   
278       3   7.0      4      1   29.1250       1           0           1   
87        3  40.0      0      0    8.0500       1           0           0   
735       3  28.5      0      0   16.1000       1           0           0   
887       1  19.0      0      0   30.0000       0           0           0   
863       3  35.0      8      2   69.5500       0           0           0   

     Embarked_S  Age_scaled  Fare_scaled  
689           1   -1.206317     3.585812  
135           0   -0.615140    -0.349461  
97            0   -0.615140     0.619112  
25            1    0.493317    -0.021841  
424           1   -0.984625    -0.245879  
..          ...         ...          ...  
278           0   -1.797494    -0.067200  
87            1    0.641112    -0.489714  
735           1   -0.208705    -0.328327  
887           1   -0.910728    -0.049658  
863           1    0.271626     0.743244  

[616 rows x 11 columns]

In [ ]:
X_test

Pclass   Age  SibSp  Parch     Fare  Sex_le  Embarked_C  Embarked_Q  \
506       2  33.0      0      2  26.0000       0           0           0   
792       3  35.0      8      2  69.5500       0           0           0   
384       3  40.0      0      0   7.8958       1           0           0   
509       3  26.0      0      0  56.4958       1           0           0   
805       3  31.0      0      0   7.7750       1           0           0   
..      ...   ...    ...    ...      ...     ...         ...         ...   
766       1  40.0      0      0  39.6000       1           1           0   
358       3  35.0      0      0   7.8792       0           0           1   
230       1  35.0      1      0  83.4750       0           0           0   
197       3  42.0      0      1   8.4042       1           0           0   
151       1  22.0      1      0  66.6000       0           0           0   

     Embarked_S  Age_scaled  Fare_scaled  
506           1    0.116244    -0.131834  
792           1    0.263225     0.740835  
384           1    0.630679    -0.494612  
509           1   -0.398192     0.479251  
805           1   -0.030738    -0.497033  
..          ...         ...          ...  
766           0    0.630679     0.140687  
358           0    0.263225    -0.494945  
230           1    0.263225     1.019868  
197           1    0.777661    -0.484425  
151           1   -0.692155     0.681722  

[264 rows x 11 columns]

In [ ]:
X_train.info()

<class 'pandas.core.frame.DataFrame'>
Index: 616 entries, 689 to 863
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Pclass       616 non-null    int64  
 1   Age          616 non-null    float64
 2   SibSp        616 non-null    int64  
 3   Parch        616 non-null    int64  
 4   Fare         616 non-null    float64
 5   Sex_le       616 non-null    int64  
 6   Embarked_C   616 non-null    int64  
 7   Embarked_Q   616 non-null    int64  
 8   Embarked_S   616 non-null    int64  
 9   Age_scaled   616 non-null    float64
 10  Fare_scaled  616 non-null    float64
dtypes: float64(4), int64(7)
memory usage: 57.8 KB


In [ ]:
X_test.info()

<class 'pandas.core.frame.DataFrame'>
Index: 264 entries, 506 to 151
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Pclass       264 non-null    int64  
 1   Age          264 non-null    float64
 2   SibSp        264 non-null    int64  
 3   Parch        264 non-null    int64  
 4   Fare         264 non-null    float64
 5   Sex_le       264 non-null    int64  
 6   Embarked_C   264 non-null    int64  
 7   Embarked_Q   264 non-null    int64  
 8   Embarked_S   264 non-null    int64  
 9   Age_scaled   264 non-null    float64
 10  Fare_scaled  264 non-null    float64
dtypes: float64(4), int64(7)
memory usage: 24.8 KB


Converta os parâmetros padronizados em um DataFrame e verifique o cabeçalho desse DataFrame para garantir que a transform() funcionou.

In [ ]:
fare_scaler = StandardScaler()
fare_scaler.fit(X_train[['Fare']])

StandardScaler()

In [ ]:
fare_scaler.transform(X_train[['Fare']])

array([[ 3.58581199e+00],
       [-3.49461306e-01],
       [ 6.19111950e-01],
       [-2.18414644e-02],
       [-2.45878849e-01],
       [-5.06169191e-01],
       [-5.05752191e-01],
       [-4.93137934e-01],
       [-5.09846010e-01],
       [-4.92805135e-01],
       [-4.97732956e-01],
       [ 1.30774606e-01],
       [ 7.43243697e-01],
       [-2.40968743e-02],
       [-4.89713720e-01],
       [-4.96145147e-01],
       [ 1.01606409e+00],
       [ 1.38793841e-01],
       [-5.08258202e-01],
       [-2.94244861e-01],
       [-4.93137934e-01],
       [-4.99737764e-01],
       [-4.95728147e-01],
       [-4.77434266e-01],
       [ 9.52746218e-01],
       [ 1.44557667e-01],
       [ 8.87172931e-01],
       [-3.60403552e-01],
       [-4.95728147e-01],
       [-1.89994803e-01],
       [-4.94809944e-01],
       [-4.92805135e-01],
       [-5.08258202e-01],
       [-4.92219731e-01],
       [ 6.84101838e-01],
       [ 2.42552889e+00],
       [-4.60643993e-01],
       [-4.99737764e-01],
       [-5.0

In [ ]:
fare_scaler.inverse_transform(df_embarked[['Fare']].head(5))

array([[ 394.10744769],
       [3588.09281444],
       [ 427.77649368],
       [2681.10857156],
       [ 434.0115022 ]])

## Conclusão

O presente trabalho teve o objetivo de realizar uma breve análise exploratória dos dados para compreensão inicial, transformações e tratamento dos dados, assim como proceder com a divisão dos dados em duas partes: treino e teste.

Os dados foram carregados, as bibliotecas necessárias foram importadas, os dados explorados e transformados (categoricos/binarização).

Na parte da exploração dos dados (EDA), foi possível verificar que as colunas Age, Embarked e Cabin tinham dados bem discrepantes dos demais, o que tornou imprescindível a limpeza e padronização dos mesmos.

Inicialmente em "Age", foi utilizada a mediana das categorias classe e sexo para aplicar nos dados ausentes e posteriormente foram excluídos os dados que não permitiram essa padronização. A coluna "Cabin" foi excluida devido a ausência de interesse em realizar análises, bem como devido ao grande número de dados faltantes. E a coluna Embarked foi excluída após binarização.

Na parte da categorização dos dados, foi utilizada a coluna "Sex", via procedimento Label Encoder, de modo que male foi transformado em 1 e female em 0. Na binarização, por sua vez, foi utilizada a coluna 'Embarked', o que permitiu verificar via binários os portões que os passageiros embarcaram.

No total, após a transformação, exclusão e tratamento dos dados nulos, ausentes, o dataset permaneceu com 11 colunas mas diferentes e o número total de dados ficou 880.

A separação dos dados em treino (616) e teste (264) é importante para aplicação de modelos e analises dos dados do ponto de vista do seu balanceamento e comportamento.

Por fim, os dados foram normalizados, o que permitiu padronizar as colunas.